### ISSUE TITLE
NoRest and FullRest is not working as expected on appium.webdriver

### ISSUE THREAD

**Issue by sheikhjebran** [appium/appium#13299]
## The problem

Expected Scenario: I want to run test one by one to which I don't want to uninstall the app or clear the app data. hence am using the below capabilities

appiumOptions.AddAdditionalCapability(MobileCapabilityType.FullReset, false);
appiumOptions.AddAdditionalCapability(MobileCapabilityType.NoReset, true);

## Environment
Appium --version . 1.15.0
* Last Appium version that did not exhibit the issue (if applicable): 1.12.1
* Desktop OS/version used to run Appium:   MAC mojave Os version: 10.14.6 (18G95)
* Node.js version (unless using Appium.app|exe):v11.2.0
* Npm or Yarn package manager:6.11.3
* Mobile platform/version under test: IOS OS:13.0
* Real device or emulator/simulator: Simulator

## Details
The above capabilities used to work as expected in appium 1.12.1 but when I upgraded the Xcode and appium they stopped working, and it reset the app for every test

## Code To Reproduce Issue [ Good To Have ]

            var appiumOptions = new AppiumOptions();
            appiumOptions.AddAdditionalCapability(MobileCapabilityType.FullReset, false);
            appiumOptions.AddAdditionalCapability(MobileCapabilityType.NoReset, true);

**Comment by mykola-mokhnach** [comment]
Please provide the full server log and the logs comparison between the working and non working releases

**Comment by sheikhjebran** [comment]
Present Logs

`[HTTP] --> POST /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244/appium/device/remove_app
[HTTP] {"appId":"com.vanquis.evanquis.dev"}
[debug] [W3C (35ba7d13)] Calling AppiumDriver.removeApp() with args: ["com.vanquis.evanquis.dev",null,null,null,"35ba7d13-e81c-4135-8d35-b166e21ab244"]
[debug] [XCUITest] Executing command 'removeApp'
[XCUITest] Uninstalling the application with bundle identifier 'com.vanquis.evanquis.dev' from the Simulator with UDID C94DD246-CB48-4D73-AA79-F280A2EFFE52
[XCUITest] Removal of 'com.vanquis.evanquis.dev' succeeded
[debug] [W3C (35ba7d13)] Responding to client with driver.removeApp() result: true
[HTTP] <-- POST /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244/appium/device/remove_app 200 1153 ms - 14
[HTTP] 
[HTTP] --> POST /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244/appium/device/install_app
[HTTP] {"appPath":"/Users/jebran/Desktop/Simlator/Vanquis.Mobile.Platform.iOS.app"}
[debug] [W3C (35ba7d13)] Calling AppiumDriver.installApp() with args: ["/Users/jebran/Desktop/Simlator/Vanquis.Mobile.Platform.iOS.app",null,null,null,"35ba7d13-e81c-4135-8d35-b166e21ab244"]
[debug] [XCUITest] Executing command 'installApp'
[BaseDriver] Using local app '/Users/jebran/Desktop/Simlator/Vanquis.Mobile.Platform.iOS.app'
[XCUITest] Installing '/Users/jebran/Desktop/Simlator/Vanquis.Mobile.Platform.iOS.app' to the Simulator with UDID C94DD246-CB48-4D73-AA79-F280A2EFFE52
[simctl] Error: simctl error running 'install': An error was encountered processing the command (domain=IXUserPresentableErrorDomain, code=1):
[simctl] This app could not be installed at this time.
[simctl] Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[simctl] Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[simctl] Underlying error (domain=MIInstallerErrorDomain, code=35):
[simctl] 	Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[simctl]     at Object.wrappedLogger.errorAndThrow (/usr/local/lib/node_modules/appium/node_modules/appium-support/lib/logging.js:79:13)
[simctl]     at errorAndThrow (/usr/local/lib/node_modules/appium/node_modules/node-simctl/lib/simctl.js:56:11)
[debug] [W3C (35ba7d13)] Encountered internal error running command: Error: simctl error running 'install': An error was encountered processing the command (domain=IXUserPresentableErrorDomain, code=1):
[debug] [W3C (35ba7d13)] This app could not be installed at this time.
[debug] [W3C (35ba7d13)] Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[debug] [W3C (35ba7d13)] Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[debug] [W3C (35ba7d13)] Underlying error (domain=MIInstallerErrorDomain, code=35):
[debug] [W3C (35ba7d13)] 	Failed to load Info.plist from bundle at path /Users/jebran/Library/Developer/CoreSimulator/Devices/C94DD246-CB48-4D73-AA79-F280A2EFFE52/data/Library/Caches/com.apple.mobile.installd.staging/temp.cQgPZS/extracted/Payload/Vanquis.Mobile.Platform.iOS.app
[debug] [W3C (35ba7d13)]     at Object.wrappedLogger.errorAndThrow (/usr/local/lib/node_modules/appium/node_modules/appium-support/lib/logging.js:79:13)
[debug] [W3C (35ba7d13)]     at errorAndThrow (/usr/local/lib/node_modules/appium/node_modules/node-simctl/lib/simctl.js:56:11)
[HTTP] <-- POST /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244/appium/device/install_app 500 1401 ms - 2491
[HTTP] 
[HTTP] --> DELETE /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244
[HTTP] {}
[debug] [W3C (35ba7d13)] Calling AppiumDriver.deleteSession() with args: ["35ba7d13-e81c-4135-8d35-b166e21ab244"]
[debug] [BaseDriver] Event 'quitSessionRequested' logged at 1569573939368 (09:45:39 GMT+0100 (British Summer Time))
[Appium] Removing session 35ba7d13-e81c-4135-8d35-b166e21ab244 from our master session list
[debug] [WD Proxy] Matched '/session/35ba7d13-e81c-4135-8d35-b166e21ab244' to command name 'deleteSession'
[debug] [WD Proxy] Proxying [DELETE /session/35ba7d13-e81c-4135-8d35-b166e21ab244] to [DELETE http://localhost:8100/session/D48DFD78-483A-45CE-BF3E-D08E38CABA6A] with no body
[debug] [WD Proxy] Got response with status 200: {
[debug] [WD Proxy]   "value" : null,
[debug] [WD Proxy]   "sessionId" : "7151F322-9A24-4A00-9C9F-BA8F1676A10B"
[debug] [WD Proxy] }
[XCUITest] Shutting down sub-processes
[XCUITest] Shutting down xcodebuild process (pid 10045)
[XCUITest] xcodebuild exited with code 'null' and signal 'SIGTERM'
[DevCon Factory] Releasing connections for C94DD246-CB48-4D73-AA79-F280A2EFFE52 device on any port number
[DevCon Factory] Found cached connections to release: ["C94DD246-CB48-4D73-AA79-F280A2EFFE52:8100"]
[debug] [DevCon Factory] Cached connections count: 0
[debug] [XCUITest] Not clearing log files. Use `clearSystemFiles` capability to turn on.
[debug] [IOSSimulatorLog] Stopping iOS log capture
[debug] [BaseDriver] Event 'quitSessionFinished' logged at 1569573939400 (09:45:39 GMT+0100 (British Summer Time))
[debug] [W3C (35ba7d13)] Received response: null
[debug] [W3C (35ba7d13)] But deleting session, so not returning
[debug] [W3C (35ba7d13)] Responding to client with driver.deleteSession() result: null
[HTTP] <-- DELETE /wd/hub/session/35ba7d13-e81c-4135-8d35-b166e21ab244 200 33 ms - 14
[HTTP] 
`

**Comment by mykola-mokhnach** [comment]
The error is coming from the Simulator itself. Try to reset it and restart your test

**Comment by sheikhjebran** [comment]
I uninstalled and re-installed but the issue is still same

**Comment by mykola-mokhnach** [comment]
Did you reset the simulator?

**Comment by sheikhjebran** [comment]
@mykola-mokhnach yes i did reset the simulator, in-fact, I completely deleted the simulator and Xcode from mac and reinstalled the fresh copy .


This issue started after updating the xcode to 11 and appium to 1.15.0 

1. unable to install the app
2. fullrest and norest are not working

**Comment by mykola-mokhnach** [comment]
We did several fixes to appium@beta
You can try if it works for you

**Comment by sheikhjebran** [comment]
Good, May I know the beta version. please

**Comment by mykola-mokhnach** [comment]
Ponting npm to `appium@beta` will always install the most recent one

**Comment by sheikhjebran** [comment]
@mykola-mokhnach Thank you will try as per the above comment and let you know

**Comment by sheikhjebran** [comment]
@mykola-mokhnach  Tried with Beta version but the NoReset and Full reset is not working as expected,

Every time the app data is getting clear for each test for ios device, but the same code works fine for android device

**Comment by sheikhjebran** [comment]
Hi @mykola-mokhnach , Used the Beta version, and finally got clarification,

The NoRest and FullReset is working find,

But the actual issue is, the appium is clearing the app data once the test is complete, 

It's like after the each test the appium is clearing the app data automatically.

**Comment by mykola-mokhnach** [comment]
Please create a new issue for it and attach the necessary logs to investigate along with extract steps to reproduce and the expected result
